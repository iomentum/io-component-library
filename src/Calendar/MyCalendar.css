.dayz {
    height: 500px;
    font-size: 1rem;
    font-family: "Roboto", "Helvetica", "Arial", sans-serif;
    font-weight: 400;
    letter-spacing: 0.00938em;
}
