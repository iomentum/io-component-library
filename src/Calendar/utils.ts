import * as m from 'moment';
import { DateRange, extendMoment, MomentRangeStaticMethods } from 'moment-range';
import Dayz from 'dayz';

export const extendedMoment = extendMoment(m);

export type MomentRangeExtended = MomentRangeStaticMethods & m.Moment;

export enum DisplayMode {
  Week = 'week',
  Day = 'day',
}

export interface Event {
  content: string;
  range: () => DateRange;
}

export interface EventsCollection {
  events: Event[];
  add: (
    eventAttrs: {
      content: string;
      range: DateRange;
    },
    options?: Record<string, never>
  ) => void;
}

export const hours = [
  '00:00',
  '00:15',
  '00:30',
  '00:45',
  '01:00',
  '01:15',
  '01:30',
  '01:45',
  '02:00',
  '02:15',
  '02:30',
  '02:45',
  '03:00',
  '03:15',
  '03:30',
  '03:45',
  '04:00',
  '04:15',
  '04:30',
  '04:45',
  '05:00',
  '05:15',
  '05:30',
  '05:45',
  '06:00',
  '06:15',
  '06:30',
  '06:45',
  '07:00',
  '07:15',
  '07:30',
  '07:45',
  '08:00',
  '08:15',
  '08:30',
  '08:45',
  '09:00',
  '09:15',
  '09:30',
  '09:45',
  '10:00',
  '10:15',
  '10:30',
  '10:45',
  '11:00',
  '11:15',
  '11:30',
  '11:45',
  '12:00',
  '12:15',
  '12:30',
  '12:45',
  '13:00',
  '13:15',
  '13:30',
  '13:45',
  '14:00',
  '14:15',
  '14:30',
  '14:45',
  '15:00',
  '15:15',
  '15:30',
  '15:45',
  '16:00',
  '16:15',
  '16:30',
  '16:45',
  '17:00',
  '17:15',
  '17:30',
  '17:45',
  '18:00',
  '18:15',
  '18:30',
  '18:45',
  '19:00',
  '19:15',
  '19:30',
  '19:45',
  '20:00',
  '20:15',
  '20:30',
  '20:45',
  '21:00',
  '21:15',
  '21:30',
  '21:45',
  '22:00',
  '22:15',
  '22:30',
  '22:45',
  '23:00',
  '23:15',
  '23:30',
  '23:45',
];

export const fromDateForHours = (date: Date) => {
  const min = date.getMinutes();

  if (min < 15) {
    return hours[date.getHours() * 4];
  }
  if (min < 30) {
    return hours[date.getHours() * 4 + 1];
  }
  if (min < 45) {
    return hours[date.getHours() * 4 + 2];
  }
  return hours[date.getHours() * 4 + 3];
};

export const getDefaultEvent = (start: MomentRangeExtended): Event => ({
  content: '',
  range: () => extendedMoment.range(extendedMoment(start), extendedMoment(start).add(1, 'hour')),
});

interface SelectedEvent {
  content: string;
  range: DateRange;
}

export const findEvent = (event: SelectedEvent, events: EventsCollection) =>
  events.events.find(
    (currEvent) => event.content === currEvent.content && event.range.isSame(currEvent.range())
  );

// const date = extendedMoment();

export const EVENTS: EventsCollection = new Dayz.EventsCollection([
  // {
  //   content: "Weeklong",
  //   range: extendedMoment.range(date.clone(), date.clone().endOf("day")),
  // },
  // {
  //   content: "9am - 2pm",
  //   range: extendedMoment.range(date.clone().hour(9), date.clone().hour(14)),
  // },
  // {
  //   content: "8am - 8pm",
  //   range: extendedMoment.range(date.clone().hour(8), date.clone().hour(21).minutes(40)),
  // },
]);
