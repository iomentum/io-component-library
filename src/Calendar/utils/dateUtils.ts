export const getFormattedDate = (date: Date) => date.toISOString().replace(/T.*/, '');

export const formatDateAndHour = (dateToFormat: Date): [string, string] => {
  const timeZoneOffset = dateToFormat.getTimezoneOffset() * 60000;
  const [date, hour] = new Date(dateToFormat.getTime() - timeZoneOffset).toISOString().split('T');
  return [date, hour.slice(0, 5)];
};

export const mergeDateAndHour = (date: Date, hour: string) => {
  const [hours, minutes] = hour.split(':');
  return new Date(date.setHours(+hours, +minutes));
};
