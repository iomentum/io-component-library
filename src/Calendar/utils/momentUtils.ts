import moment, * as m from 'moment';
import { MomentInput } from 'moment';
import { extendMoment } from 'moment-range';

const extendedMoment = extendMoment(m);

export const createExtendedMomentFromDate = (date: Date) => extendedMoment(date);

export const createMomentRange = (startDate: Date, endDate: Date) =>
  extendedMoment.range(extendedMoment(new Date(startDate)), extendedMoment(new Date(endDate)));

export const createDateFromMoment = (momentInput: MomentInput) => moment(momentInput).toDate();
