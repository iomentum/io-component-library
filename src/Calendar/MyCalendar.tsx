import React, { useMemo, useState } from 'react';
import { CalendarHeader } from './components/CalendarHeader';
import { EventManagement } from './components/EventManagement';
import {
  getDefaultEvent,
  EVENTS,
  MomentRangeExtended,
  extendedMoment,
  Display,
  EventsCollection,
} from './utils';

import 'dayz/dist/dayz.css';
import './MyCalendar.css';
import { CalendarContext, CalendarContextInterface } from './contexts/CalendarContext';
import { EventContext, EventContextInterface } from './contexts/EventContext';
import { DayzWrapper } from './components/DayzWrapper';

export interface CalendarProps {
  displayedDate: MomentRangeExtended;
  displayMode: Display;
  events?: EventsCollection;
}

export function MyCalendar(props: CalendarProps) {
  const [displayedDate, setDisplayedDate] = useState(props.displayedDate);
  const [displayMode, setDisplayMode] = useState(props.displayMode);
  const [eventsCollection, setEventsCollection] = useState(props.events);
  const [eventManagementOpened, setEventManagementOpened] = useState(false);
  const [currentEvent, setCurrentEvent] = useState(getDefaultEvent(extendedMoment()));

  const calendarContextValue = useMemo<CalendarContextInterface>(
    () => ({
      displayMode,
      setDisplayMode,
      displayedDate,
      setDisplayedDate,
    }),
    [displayMode, displayedDate]
  );

  const eventContextValue = useMemo<EventContextInterface>(
    () => ({
      eventsCollection,
      setEventsCollection,
      currentEvent,
      setCurrentEvent,
      eventManagementOpened,
      setEventManagementOpened,
    }),
    [eventsCollection, currentEvent, eventManagementOpened]
  );

  return (
    <>
      <CalendarContext.Provider value={calendarContextValue}>
        <CalendarHeader />
        <EventContext.Provider value={eventContextValue}>
          <DayzWrapper />
          <EventManagement />
        </EventContext.Provider>
      </CalendarContext.Provider>
    </>
  );
}

MyCalendar.defaultProps = {
  events: EVENTS,
};
