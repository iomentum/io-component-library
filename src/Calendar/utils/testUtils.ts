import { Event } from '../reducers/EventReducer';
import { formatDateAndHour } from './dateUtils';

const dates = [
  { start: new Date('2021-08-05T11:00:00'), end: new Date('2021-08-05T12:00:00') },
  { start: new Date('2021-08-05T14:00:00'), end: new Date('2021-08-05T16:00:00') },
  { start: new Date('2021-08-05T18:00:00'), end: new Date('2021-08-05T19:00:00') },
];

export const createMockedEvents: Event[] = dates.map((date, index) => {
  const [displayStartDate, startHour] = formatDateAndHour(date.start);
  const [displayEndDate, endHour] = formatDateAndHour(date.end);

  return {
    uuid: `awdaw1-123${index}`,
    title: `test ${index}`,
    startDate: date.start,
    endDate: date.end,
    displayStartDate,
    displayEndDate,
    startHour,
    endHour,
    metadata: {},
  };
});

export const getMockedEvent = () =>
  createMockedEvents.find((event) => event.uuid === `awdaw1-1231`);
