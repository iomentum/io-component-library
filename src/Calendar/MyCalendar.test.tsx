/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import { MyCalendar } from './MyCalendar';
import { DisplayMode, Event } from './utils';

const createEvent = (eventStart: Date): Event => {
  const eventEnd = new Date(eventStart);
  eventEnd.setHours(eventEnd.getHours() + 1);

  return {
    content: '',
    dateRange: {
      eventStart,
      eventEnd,
    },
  };
};

const formatDateAndHour = (dateToFormat: Date): [string, string] => {
  const timeZoneOffset = dateToFormat.getTimezoneOffset() * 60000;
  const [date, hour] = new Date(dateToFormat.getTime() - timeZoneOffset).toISOString().split('T');
  return [date, hour.slice(0, 5)];
};

jest.mock('./utils', () => ({
  computeIsFullDayEvent: jest.fn(),
  createExtendedMomentFromDate: () => new Date('2021-08-03'),
  DisplayMode: 'week',
  createEvent: () => createEvent(new Date('2021-08-03')),
  formatDateAndHour: () => formatDateAndHour(new Date('2021-08-03')),
}));

jest.mock('./utils/eventUtils', () => ({ isEventExisting: jest.fn() }));

describe('MyCalendar component', () => {
  describe('@snapshots', () => {
    it('should match with previous MyCalendar Month', () => {
      const newDate: Date = new Date('2021-08-03');
      const { asFragment } = render(
        <MyCalendar displayMode={DisplayMode.Month} displayedDate={newDate} />
      );

      expect(asFragment()).toMatchSnapshot('MyCalendar snapshot Month');
    });

    it('should match with previous MyCalendar Week', () => {
      const newDate: Date = new Date('2021-08-03');
      const { asFragment } = render(
        <MyCalendar displayMode={DisplayMode.Week} displayedDate={newDate} />
      );

      expect(asFragment()).toMatchSnapshot('MyCalendar snapshot Week');
    });

    it('should match with previous MyCalendar Day', () => {
      const newDate: Date = new Date('2021-08-03');
      const { asFragment } = render(
        <MyCalendar displayMode={DisplayMode.Day} displayedDate={newDate} />
      );

      expect(asFragment()).toMatchSnapshot('MyCalendar snapshot Day');
    });
  });
});
