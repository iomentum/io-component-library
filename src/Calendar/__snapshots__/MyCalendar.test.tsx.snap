// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCalendar component @snapshots should match with previous MyCalendar Day: MyCalendar snapshot Day 1`] = `
<DocumentFragment>
  <div
    class="tools"
  >
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-8"
            data-indeterminate="false"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        W
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-8"
            data-indeterminate="false"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        E
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-8"
            data-indeterminate="false"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        E
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-8"
            data-indeterminate="false"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        K
      </span>
    </label>
  </div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      &lt;
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Today
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      &gt;
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <div
    class="dayz month"
  >
    <div
      class="x-labels"
    >
      <div
        class="day-label"
      >
        Sunday
      </div>
      <div
        class="day-label"
      >
        Monday
      </div>
      <div
        class="day-label"
      >
        Tuesday
      </div>
      <div
        class="day-label"
      >
        Wednesday
      </div>
      <div
        class="day-label"
      >
        Thursday
      </div>
      <div
        class="day-label"
      >
        Friday
      </div>
      <div
        class="day-label"
      >
        Saturday
      </div>
    </div>
    <div
      class="body"
    >
      <div
        class="days"
      >
        <div
          class="day before"
          data-date="20210801"
          style="order: 0;"
        >
          <div
            class="label"
          >
            1
          </div>
        </div>
        <div
          class="day before"
          data-date="20210802"
          style="order: 1;"
        >
          <div
            class="label"
          >
            2
          </div>
        </div>
        <div
          class="day before"
          data-date="20210803"
          style="order: 2;"
        >
          <div
            class="label"
          >
            3
          </div>
        </div>
        <div
          class="day after"
          data-date="20210804"
          style="order: 3;"
        >
          <div
            class="label"
          >
            4
          </div>
        </div>
        <div
          class="day after"
          data-date="20210805"
          style="order: 4;"
        >
          <div
            class="label"
          >
            5
          </div>
        </div>
        <div
          class="day after"
          data-date="20210806"
          style="order: 5;"
        >
          <div
            class="label"
          >
            6
          </div>
        </div>
        <div
          class="day after"
          data-date="20210807"
          style="order: 6;"
        >
          <div
            class="label"
          >
            7
          </div>
        </div>
        <div
          class="day after"
          data-date="20210808"
          style="order: 7;"
        >
          <div
            class="label"
          >
            8
          </div>
        </div>
        <div
          class="day after"
          data-date="20210809"
          style="order: 8;"
        >
          <div
            class="label"
          >
            9
          </div>
        </div>
        <div
          class="day after"
          data-date="20210810"
          style="order: 9;"
        >
          <div
            class="label"
          >
            10
          </div>
        </div>
        <div
          class="day after"
          data-date="20210811"
          style="order: 10;"
        >
          <div
            class="label"
          >
            11
          </div>
        </div>
        <div
          class="day after"
          data-date="20210812"
          style="order: 11;"
        >
          <div
            class="label"
          >
            12
          </div>
        </div>
        <div
          class="day after"
          data-date="20210813"
          style="order: 12;"
        >
          <div
            class="label"
          >
            13
          </div>
        </div>
        <div
          class="day after"
          data-date="20210814"
          style="order: 13;"
        >
          <div
            class="label"
          >
            14
          </div>
        </div>
        <div
          class="day after"
          data-date="20210815"
          style="order: 14;"
        >
          <div
            class="label"
          >
            15
          </div>
        </div>
        <div
          class="day after"
          data-date="20210816"
          style="order: 15;"
        >
          <div
            class="label"
          >
            16
          </div>
        </div>
        <div
          class="day after"
          data-date="20210817"
          style="order: 16;"
        >
          <div
            class="label"
          >
            17
          </div>
        </div>
        <div
          class="day after"
          data-date="20210818"
          style="order: 17;"
        >
          <div
            class="label"
          >
            18
          </div>
        </div>
        <div
          class="day after"
          data-date="20210819"
          style="order: 18;"
        >
          <div
            class="label"
          >
            19
          </div>
        </div>
        <div
          class="day after"
          data-date="20210820"
          style="order: 19;"
        >
          <div
            class="label"
          >
            20
          </div>
        </div>
        <div
          class="day after"
          data-date="20210821"
          style="order: 20;"
        >
          <div
            class="label"
          >
            21
          </div>
        </div>
        <div
          class="day after"
          data-date="20210822"
          style="order: 21;"
        >
          <div
            class="label"
          >
            22
          </div>
        </div>
        <div
          class="day after"
          data-date="20210823"
          style="order: 22;"
        >
          <div
            class="label"
          >
            23
          </div>
        </div>
        <div
          class="day after"
          data-date="20210824"
          style="order: 23;"
        >
          <div
            class="label"
          >
            24
          </div>
        </div>
        <div
          class="day after"
          data-date="20210825"
          style="order: 24;"
        >
          <div
            class="label"
          >
            25
          </div>
        </div>
        <div
          class="day after"
          data-date="20210826"
          style="order: 25;"
        >
          <div
            class="label"
          >
            26
          </div>
        </div>
        <div
          class="day after"
          data-date="20210827"
          style="order: 26;"
        >
          <div
            class="label"
          >
            27
          </div>
        </div>
        <div
          class="day after"
          data-date="20210828"
          style="order: 27;"
        >
          <div
            class="label"
          >
            28
          </div>
        </div>
        <div
          class="day after"
          data-date="20210829"
          style="order: 28;"
        >
          <div
            class="label"
          >
            29
          </div>
        </div>
        <div
          class="day after"
          data-date="20210830"
          style="order: 29;"
        >
          <div
            class="label"
          >
            30
          </div>
        </div>
        <div
          class="day after"
          data-date="20210831"
          style="order: 30;"
        >
          <div
            class="label"
          >
            31
          </div>
        </div>
        <div
          class="day after outside"
          data-date="20210901"
          style="order: 31;"
        >
          <div
            class="label"
          >
            1
          </div>
        </div>
        <div
          class="day after outside"
          data-date="20210902"
          style="order: 32;"
        >
          <div
            class="label"
          >
            2
          </div>
        </div>
        <div
          class="day after outside"
          data-date="20210903"
          style="order: 33;"
        >
          <div
            class="label"
          >
            3
          </div>
        </div>
        <div
          class="day after outside"
          data-date="20210904"
          style="order: 34;"
        >
          <div
            class="label"
          >
            4
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MyCalendar component @snapshots should match with previous MyCalendar Week: MyCalendar snapshot Week 1`] = `
<DocumentFragment>
  <div
    class="tools"
  >
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-4"
            data-indeterminate="false"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        W
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-4"
            data-indeterminate="false"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        E
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-4"
            data-indeterminate="false"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        E
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-4"
            data-indeterminate="false"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        K
      </span>
    </label>
  </div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      &lt;
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Today
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      &gt;
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <div
    class="dayz month"
  >
    <div
      class="x-labels"
    >
      <div
        class="day-label"
      >
        Sunday
      </div>
      <div
        class="day-label"
      >
        Monday
      </div>
      <div
        class="day-label"
      >
        Tuesday
      </div>
      <div
        class="day-label"
      >
        Wednesday
      </div>
      <div
        class="day-label"
      >
        Thursday
      </div>
      <div
        class="day-label"
      >
        Friday
      </div>
      <div
        class="day-label"
      >
        Saturday
      </div>
    </div>
    <div
      class="body"
    >
      <div
        class="days"
      >
        <div
          class="day before"
          data-date="20210801"
          style="order: 0;"
        >
          <div
            class="label"
          >
            1
          </div>
        </div>
        <div
          class="day before"
          data-date="20210802"
          style="order: 1;"
        >
          <div
            class="label"
          >
            2
          </div>
        </div>
        <div
          class="day before"
          data-date="20210803"
          style="order: 2;"
        >
          <div
            class="label"
          >
            3
          </div>
        </div>
        <div
          class="day after"
          data-date="20210804"
          style="order: 3;"
        >
          <div
            class="label"
          >
            4
          </div>
        </div>
        <div
          class="day after"
          data-date="20210805"
          style="order: 4;"
        >
          <div
            class="label"
          >
            5
          </div>
        </div>
        <div
          class="day after"
          data-date="20210806"
          style="order: 5;"
        >
          <div
            class="label"
          >
            6
          </div>
        </div>
        <div
          class="day after"
          data-date="20210807"
          style="order: 6;"
        >
          <div
            class="label"
          >
            7
          </div>
        </div>
        <div
          class="day after"
          data-date="20210808"
          style="order: 7;"
        >
          <div
            class="label"
          >
            8
          </div>
        </div>
        <div
          class="day after"
          data-date="20210809"
          style="order: 8;"
        >
          <div
            class="label"
          >
            9
          </div>
        </div>
        <div
          class="day after"
          data-date="20210810"
          style="order: 9;"
        >
          <div
            class="label"
          >
            10
          </div>
        </div>
        <div
          class="day after"
          data-date="20210811"
          style="order: 10;"
        >
          <div
            class="label"
          >
            11
          </div>
        </div>
        <div
          class="day after"
          data-date="20210812"
          style="order: 11;"
        >
          <div
            class="label"
          >
            12
          </div>
        </div>
        <div
          class="day after"
          data-date="20210813"
          style="order: 12;"
        >
          <div
            class="label"
          >
            13
          </div>
        </div>
        <div
          class="day after"
          data-date="20210814"
          style="order: 13;"
        >
          <div
            class="label"
          >
            14
          </div>
        </div>
        <div
          class="day after"
          data-date="20210815"
          style="order: 14;"
        >
          <div
            class="label"
          >
            15
          </div>
        </div>
        <div
          class="day after"
          data-date="20210816"
          style="order: 15;"
        >
          <div
            class="label"
          >
            16
          </div>
        </div>
        <div
          class="day after"
          data-date="20210817"
          style="order: 16;"
        >
          <div
            class="label"
          >
            17
          </div>
        </div>
        <div
          class="day after"
          data-date="20210818"
          style="order: 17;"
        >
          <div
            class="label"
          >
            18
          </div>
        </div>
        <div
          class="day after"
          data-date="20210819"
          style="order: 18;"
        >
          <div
            class="label"
          >
            19
          </div>
        </div>
        <div
          class="day after"
          data-date="20210820"
          style="order: 19;"
        >
          <div
            class="label"
          >
            20
          </div>
        </div>
        <div
          class="day after"
          data-date="20210821"
          style="order: 20;"
        >
          <div
            class="label"
          >
            21
          </div>
        </div>
        <div
          class="day after"
          data-date="20210822"
          style="order: 21;"
        >
          <div
            class="label"
          >
            22
          </div>
        </div>
        <div
          class="day after"
          data-date="20210823"
          style="order: 22;"
        >
          <div
            class="label"
          >
            23
          </div>
        </div>
        <div
          class="day after"
          data-date="20210824"
          style="order: 23;"
        >
          <div
            class="label"
          >
            24
          </div>
        </div>
        <div
          class="day after"
          data-date="20210825"
          style="order: 24;"
        >
          <div
            class="label"
          >
            25
          </div>
        </div>
        <div
          class="day after"
          data-date="20210826"
          style="order: 25;"
        >
          <div
            class="label"
          >
            26
          </div>
        </div>
        <div
          class="day after"
          data-date="20210827"
          style="order: 26;"
        >
          <div
            class="label"
          >
            27
          </div>
        </div>
        <div
          class="day after"
          data-date="20210828"
          style="order: 27;"
        >
          <div
            class="label"
          >
            28
          </div>
        </div>
        <div
          class="day after"
          data-date="20210829"
          style="order: 28;"
        >
          <div
            class="label"
          >
            29
          </div>
        </div>
        <div
          class="day after"
          data-date="20210830"
          style="order: 29;"
        >
          <div
            class="label"
          >
            30
          </div>
        </div>
        <div
          class="day after"
          data-date="20210831"
          style="order: 30;"
        >
          <div
            class="label"
          >
            31
          </div>
        </div>
        <div
          class="day after outside"
          data-date="20210901"
          style="order: 31;"
        >
          <div
            class="label"
          >
            1
          </div>
        </div>
        <div
          class="day after outside"
          data-date="20210902"
          style="order: 32;"
        >
          <div
            class="label"
          >
            2
          </div>
        </div>
        <div
          class="day after outside"
          data-date="20210903"
          style="order: 33;"
        >
          <div
            class="label"
          >
            3
          </div>
        </div>
        <div
          class="day after outside"
          data-date="20210904"
          style="order: 34;"
        >
          <div
            class="label"
          >
            4
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
